src/map/Map.tsx

```diff
@@ -42,17 +57,69 @@ function Map() {
     longitude
   )
 
+  const mapRef = useRef<any>(null)
+
+  const [bounds, setBounds] = useState<MiniGlobeBounds | any>(null)
+
+  useEffect(() => {
+    const mapboxRef = mapRef.current && mapRef.current.getMap()
+    if (mapboxRef) {
+      const rawBounds = mapboxRef.getBounds()
+      if (rawBounds) {
+        setBounds({
+          north: rawBounds.getNorth() as number,
+          south: rawBounds.getSouth() as number,
+          west: rawBounds.getWest() as number,
+          east: rawBounds.getEast() as number,
+        })
+      }
+    }
+  }, [zoom, latitude, longitude])
+
   return (
-    <div className="map">
+    <Fragment>
       {loading && <Loader />}
       <ReactMapGL
+        ref={mapRef}
         width="100%"
         height="100%"
         {...viewport}
         onViewportChange={onViewportChange as any}
         mapStyle={style}
-      />
-    </div>
+        mapOptions={{
+          customAttribution: '© Copyright Global Fishing Watch 2019',
+        }}
+        onClick={(event) => {
+          if (rulersEditing === true) {
```

Comment:
Would be nice to start working on something generic to handle layerComposer interactions using this as base: https://github.com/GlobalFishingWatch/carrier-portal/blob/develop/src/components/map/map.tsx#L248

Reply:
You're right, it's messy. Couldn't figure out this with hooks :/